cachito/workers/pkg_managers/pip.py

```diff
@@ -1575,3 +1578,104 @@ def get_hosted_repositories_username(request_id):
     :rtype: str
     """
     return f"cachito-pip-{request_id}"
+
+
+def generate_pip_conf_content(nexus_pypi_hosted_repo_url, username, password, custom_ca_path=None):
+    """
+    Generate a pip.conf file with the nexus repo url and authentication configured.
+
+    :param str nexus_pypi_hosted_repo_url: the URL for the Nexus PyPI hosted repository
+        for the request
+    :param str username: the username of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str password: the password of the unprivileged user that has read access
+        to the main Cachito repositories
+    :param str custom_ca_path: the path to set ``cert`` to in the pip.conf file; if not provided,
+        this option will be omitted
+    :return: the contents of the pip.conf file
+    :rtype: str
+    """
+    # Instead of getting the token from Nexus, use basic authentication as supported by Nexus:
```

Comment:
Nitpick: Pip is what supports the basic authentication, not Nexus (well Nexus too, but here Pip is more relevant)